packages/node_modules/@webex/plugin-meetings/src/reconnection-manager/index.js

```diff
@@ -175,60 +175,84 @@ export default class ReconnectionManager {
     }
 
     return Promise.reject(validatedReconnect);
   }
 
   /**
-   * @param {Meeting} meeting
+   * @param {Object} reconnectOptions
+   * @param {boolean} [reconnectOptions.networkChange=false] indicates if a network change event happened
+   * @param {boolean} [reconnectOptions.networkDisconnect=false] indicates if a network disconnect event happened
    * @returns {Promise}
    * @private
    * @memberof ReconnectionManager
    */
-  execute(meeting) {
+  async executeReconnection({networkChange = false, networkDisconnect = false}) {
     this.status = RECONNECTION.STATE.IN_PROGRESS;
 
-    // Makes sure only join if you are not join else continue with reconnect
-    return new Promise((resolve, reject) => {
-      // Irrespective of who or when a reconnect is triggred try to join if in left state
-      if (meeting.state === _LEFT_) {
-        meeting.join()
-          .then(() => resolve())
-          .catch((error) => reject(error));
+
+    if (networkChange) {
+      // TODO: Calliope Discovery Needed (Out of band, no need to block execution)
+    }
+
+    if (networkDisconnect) {
+      try {
+        await this.reconnectMercuryWithBackoff();
+      }
+      catch (error) {
+        LoggerProxy.logger.error('ReconnectionManger:index#executeReconnection --> Unable to reconnect to websocket, giving up.');
+        this.status = RECONNECTION.STATE.FAILURE;
+        throw (error);
+      }
+    }
+
+    let rejoinMeeting = false;
+
+    if (this.meeting.state === _LEFT_) {
+      try {
+        rejoinMeeting = true;
+        await this.meeting.join();
+      }
+      catch (joinError) {
+        // TODO: If meeting was inactive fail the reconnect, otherwise, retry executeReconnection({networkDisconnect: true})
+        LoggerProxy.logger.error('ReconnectionManger:index#executeReconnection --> Unable to join left meeting, giving up.');
+        this.status = RECONNECTION.STATE.FAILURE;
+        throw (joinError);
       }
+    }
 
-      resolve();
-    }).then(() => this.reconnectMedia(meeting)
-      .then((media) => {
-        LoggerProxy.logger.log(`ReconnectionManager:index#execute --> Media reestablished at: ${new Date()}`);
-        this.status = RECONNECTION.STATE.COMPLETE;
+    try {
+      const media = await this.reconnectMedia();
 
-        return Promise.resolve(media);
-      })
-      .catch((err) => { // eslint-disable-line
-        LoggerProxy.logger.error(`ReconnectionManager:index#execute --> Media reestablishment failed at: ${new Date()}`);
-        if (this.tryCount > meeting.config.reconnection.retry.times) {
-          this.status = RECONNECTION.STATE.FAILURE;
+      LoggerProxy.logger.log('ReconnectionManager:index#executeReconnection --> Media reestablished');
+      this.status = RECONNECTION.STATE.COMPLETE;
 
-          return Promise.reject(new ReconnectionError(err));
-        }
-        // exponential backoff and retry logic
-        setTimeout(() =>
-          this.reconnect(meeting),
```

Comment:
I couldn't view the document for some reason, are we completely removing the reconnect backoff for media and only doing it for joins? (i saw a todo in there for backoff on the join retry)

Reply:
Correct, if the media fails we are falling back to the join request flow.

Reply:
Got it. and then if the join fails we are doing exponential backoff with a total cap limit of retries?

Reply:
Right now we are going to to just keep attempting to rejoin. At this point, we should have network connectivity, so if we get a failure message from the rejoin with an error code saying we won't be able to rejoin, it will just stop the reconnect attempt.